File: analyzers/tests/SonarAnalyzer.Rules.Tests/Cases/InsecureTemporaryFilesCreation.vb
Imports System
Imports System.IO

Public Class TestCases
    ReadOnly Property TempFileName() As String
        Get
            TempFileName = Path.GetTempFileName() 'Noncompliant
            Exit Property
        End Get
    End Property

    Public Sub Method()
        Dim TempPath = Path.GetTempFileName() 'Noncompliant
    End Sub

    Public Function GetTempFileNameGenerator() As Func(Of String)
        Return AddressOf Path.GetTempFileName 'Noncompliant
    End Function

    Public Sub Test()
        Consume(AddressOf Path.GetTempFileName) 'Noncompliant
    End Sub

    Public Sub Consume(generator as Func(Of String))
        Dim path = generator()
    End Sub

    Public Function Compliant() As String
        Return Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()) 'Compliant
    End Function
End Class

File: analyzers/tests/SonarAnalyzer.Test/TestCases/InsecureTemporaryFilesCreation.cs
using System;
using System.IO;

public class Sample
{
    private string tempFileName;

    public string TempFileName
    {
        set => tempFileName = Path.Combine(value, Path.GetTempFileName()); // Noncompliant
        get => tempFileName;
    }

    public void Method() => Path.GetTempFileName(); // Noncompliant

    public void Safe() => Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()); // Compliant
}

public class Class
{
    public Func<string> GetTempFileNameGenerator() => Path.GetTempFileName; // Noncompliant
//                                                    ^^^^^^^^^^^^^^^^^^^^

    public void Test()
    {
        Consume(Path.GetTempFileName); // Noncompliant

        _ = Custom.Path.GetTempFileName(); // Compliant, method from other namespace
    }

    private static void Consume(Func<string> generator) => generator();
}

namespace Custom
{
    public static class Path
    {
        public static string GetTempFileName() => string.Empty;
    }
}
